packages/framework/azure-client/src/test/AzureClient.spec.ts

```diff
@@ -16,47 +17,144 @@ describe("AzureClient", () => {
+     * a container successfully.
+     *
+     * Expected behavior: an error should not be thrown nor should a rejected promise
+     * be returned.
+     */
     it("can create new Azure Fluid Relay container successfully", async () => {
-        const resources = client.createContainer(schema);
+        const azureContainer = client.createContainer(schema);
 
         await assert.doesNotReject(
-            resources,
+            azureContainer,
             () => true,
             "container cannot be created in Azure Fluid Relay",
         );
-
-        const { container } = await resources;
-        assert.notStrictEqual(Object.keys(container.initialObjects), Object.keys(schema.initialObjects));
     });
 
+    /**
+     * Scenario: test when an Azure Client container is created,
+     * it is initially detached.
+     *
+     * Expected behavior: an error should not be thrown nor should a rejected promise
+     * be returned.
+     */
     it("Created container is detached", async () => {
         const { container } = await client.createContainer(schema);
         assert.strictEqual(container.attachState, AttachState.Detached, "Container should be detached");
     });
 
-    it("Can attach container", async () => {
+    /**
+     * Scenario: Test attaching a container.
+     *
+     * Expected behavior: an error should not be thrown nor should a rejected promise
+     * be returned.
+     */
+    it("can attach a container", async () => {
         const { container } = await client.createContainer(schema);
         const containerId = await container.attach();
+        await new Promise<void>((resolve) => {
+            container.on("connected", () => {
+                resolve();
+            });
+        });
 
         assert.strictEqual(
-            typeof(containerId), "string",
+            typeof (containerId), "string",
             "Attach did not return a string ID",
         );
         assert.strictEqual(
             container.attachState, AttachState.Attached,
             "Container is not attached after attach is called",
         );
+    });
+
+    /**
+     * Scenario: Test if attaching a container twice fails.
+     *
+     * Expected behavior: an error should not be thrown nor should a rejected promise
+     * be returned.
+     */
+    it("cannot attach a container twice", async () => {
+        const { container } = await client.createContainer(schema);
+        const containerId = await container.attach();
+        await new Promise<void>((resolve) => {
+            container.on("connected", () => {
+                resolve();
+            });
+        });
+
+        assert.strictEqual(
+            typeof (containerId), "string",
+            "Attach did not return a string ID",
+        );
+        assert.strictEqual(
+            container.attachState, AttachState.Attached,
+            "Container is attached after attach is called",
+        );
         await assert.rejects(
```

Comment:
were you able to hide the stack output from the tests in these cases?  if not i wonder if something like inverting the check would work?  e.g.
```
container.attach().then(() => {
    assert
}).catch(() => {
    // success case
})

Reply:
Looks like the error is actually being thrown when trying to load an improperly or non-existent container:
```
    /**
     * Scenario: test if Azure Client can get an existing container.
     *
     * Expected behavior: an error should not be thrown nor should a rejected promise
     * be returned.
     */
    it("can retrieve existing Azure Fluid Relay container successfully", async () => {
        const { container: newContainer } = await client.createContainer(schema);
        const containerId = await newContainer.attach();
        await new Promise<void>((resolve) => {
            newContainer.on("connected", () => {
                resolve();
            });
        });

        const resources = client.getContainer(containerId, schema);
        await assert.doesNotReject(
            resources,
            () => true,
            "container cannot be retrieved from Azure Fluid Relay",
        );
    });
```

Reply:
I'll try to see if I can hide the stack output for this block

Reply:
Seems like the error is being thrown from somewhere inside the logic of `client.getContainer("containerConfig", schema)`. The only way I have found to disable the error message is by adding disabling the `console.error` function before the test. I could add something like this:
```
const consoleErrorFn = console.error;
        console.error = () => {};
        const containerAndServicesP = client.getContainer("containerConfig", schema);

        const errorFn = (error) => {
            assert.notStrictEqual(error.message, undefined, "Azure Client error is undefined");
            return true;
        };

        await assert.rejects(
            containerAndServicesP,
            errorFn,
            "Azure Client can load a non-existent container",
        );
        console.error = consoleErrorFn;
```
I imagine this isn't the best practice so let me know if you have any alternative ideas or think I should just not include these changes @heliocliu

Reply:
Yeah this looks pretty questionable but at the same time this is just a test and it's relatively harmless.  I'd be fine with it lol